staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder.go

```diff
@@ -96,33 +96,39 @@ func BuildSwagger(crd *apiextensions.CustomResourceDefinition, version string) (
 
 	routes := make([]*restful.RouteBuilder, 0)
 	root := fmt.Sprintf("/apis/%s/%s/%s", b.group, b.version, b.plural)
+
+	var params []*restful.Parameter
+
 	if b.namespaced {
-		routes = append(routes, b.buildRoute(root, "", "GET", "list", sampleList).
-			Operation("list"+b.kind+"ForAllNamespaces"))
+		namespaceParam := b.ws.PathParameter("namespace", "object name and auth scope, such as for teams and projects").DataType("string")
+		routes = append(routes, b.buildRoute(root, "", "GET", "list", "list", sampleList, params).Operation("list"+b.kind+"ForAllNamespaces"))
+		params = append(params, namespaceParam)
```

Comment:
depending on a specific order of buildRoute and params building seems fragile. since the path building and `buildRoute` are both private, something like this would be easier to follow and less fragile:

local private package variables:
```
var (
  nameToken = "{name}"
  namespaceToken = "{namespace}"
)
```

in buildRoute:
```
var params []*restful.Parameter
if strings.Contains(root, namespaceToken) || strings.Contains(path, namespaceToken) {
  params = append(params, b.ws.PathParameter("namespace", "object name and auth scope, such as for teams and projects").DataType("string"))
}
if strings.Contains(root, nameToken) || strings.Contains(path, nameToken) {
  params = append(params, b.ws.PathParameter("name", "name of the "+b.kind).DataType("string"))
}
```

Reply:
refactored, ptal